### DISCUSSION TITLE
Task shadow name not used in task logger

### DISCUSSION THREAD

**Discussion by sverdoux** [celery/celery#8670] [Q&A]
Hello,

I'm am currently trying to write a common errback for various celery chords. What I mean is that I create various types of chords which will differ in their header group, but I would link_error them to the same errback task. Works fine for now but I'm trying to improve logs. What I want is to hide this errback task name from the logs because it is very generic and use instead the name of the header tasks (they are all the same). To that end i'm using the 'shadow' option but it seems from this [piece of code](https://github.com/celery/celery/blob/fb8b3caeafc15a6d00f9f06be336a3ca10c6ebdb/celery/app/log.py#L37) that it is not used in the default TaskFormatter, which surprises me because it is well used in task tracing (meaning for example that the worker logger displays properly '{shadow_name} received')
I did an ugly monkey patching to reach my goal for now but is it wanted to ignore the shadow name in task logging ?
```py
from celery.utils.log import ColorFormatter
from celery._state import get_current_task


class CustomTaskFormatter(ColorFormatter):
    """Formatter for celery tasks which uses task shadow name if available"""

    def format(self, record):
        task = get_current_task()
        if task and task.request:
            record.__dict__.update(task_id=task.request.id,
                                   task_name=getattr(task.request, 'shadow', None) or task.name)
        else:
            record.__dict__.setdefault('task_name', '???')
            record.__dict__.setdefault('task_id', '???')
        return super().format(record)


import celery.app.log  # noqa: E402
celery.app.log.TaskFormatter = CustomTaskFormatter

```

**Comment by ShaheedHaque** [comment]
I'd like to see this addressed too.

(However, shouldn't the patch above actually use `celery.app.trace.get_task_name()` for complete consistency?)

Also, I find it annoying there are inconsistent use of "Task", "[]" and "()". For example:

```
2024-01-04 11:16:11,831 [INFO] MainProcess: Task paiyroll.tasks.viewflow_dequeue[47849a7b-9a44-48a4-8a4f-1211fd761897] received
```

versus

```
2024-01-04 11:16:15,305 [INFO] SpawnPoolWorker-2:paiyroll.tasks.viewflow_dequeue(47849a7b-9a44-48a4-8a4f-1211fd761897):
```
